cloudformation-telemetry/lambda_function.py

```diff
@@ -0,0 +1,193 @@
+                self.all_resource_statuses[resourceID].append(status)
+                log_data = {
+                    'requestid': self.context.aws_request_id,
+                    'timestamp': stack_resource['Timestamp'].isoformat(timespec='milliseconds'),
+                    'data': {
+                        'stackId': self.event['StackId'],
+                        'resourceType': stack_resource["ResourceType"],
+                        'resourceName': stack_resource["LogicalResourceId"],
+                        'resourceID': stack_resource["PhysicalResourceId"],
+                        'status': stack_resource["ResourceStatus"],
+                        'details': resource_status_reason
+                    }
+                }
+                log_data_list.append(log_data)
+        return log_data_list
+
+    def fetch_and_send_telemetry(self):
+        resources_in_progress = True
+        while (resources_in_progress):
+            resources_in_progress = self._has_any_child_resources_in_progress_state()
+            log_data_list = self._create_telemetry_data()
+            for log_data in log_data_list:
+                self.send_telemetry(log_data,self.event['ResourceProperties']['TelemetryEndpoint'])
+            # If all child resources are completed except PrimeInvoker, marking PrimeInvoker as completed
+            if not resources_in_progress: 
+                helper._cfn_response(self.event)
+            time.sleep(int(self.event['ResourceProperties']['scanInterval']))
+        # If all resources are completed, make final call to know Parent stack status
+        if not resources_in_progress :
+            log_data_list = self._create_telemetry_data()
+            for log_data in log_data_list:
+                print(log_data)
+                self.send_telemetry(log_data,self.event['ResourceProperties']['TelemetryEndpoint'])
+
+
+class awsoTelemetry(parentStackTelemetry):
+
+    def __init__(self, event, context):
+        self.event = event
+        self.context = context
+        self.stackID = event['ResourceProperties']['stackID']
+        self.cfclient = boto3.client('cloudformation')
+        self.all_resource_statuses=defaultdict(list)
+    
+    def enrich_telemetry_data(self, log_data_list):
+        static_data = {
+            'profile': {
+                'sumo': {
+                    'deployment': self.event['ResourceProperties']['sumoDeployment'],
+                    'orgid': self.event['ResourceProperties']['sumoOrgId'],
+                },
+                'solution': {
+                    'name': self.event['ResourceProperties']['solutionName'],
+                    'version': self.event['ResourceProperties']['solutionVersion'],
+                },
+            }
+        }
+        for log_data in log_data_list:
+            log_data.update(static_data)
+        return log_data_list   
+
+    def fetch_and_send_telemetry(self):
```

Comment:
we do not need fetch_and_send_telemetry here, only enrich_telemetry_data needs to be overridden

Reply:
Implemented